robot_smach_states/src/robot_smach_states/navigation/navigation.py

```diff
@@ -39,6 +40,45 @@ def execute(self, userdata=None):
         return 'done'
 
 
+class PrepareDrive(smach.State):
+    """
+    Check if the robot is ready to start driving
+    """
+    def __init__(self, robot, timeout=10, reset_head=True):
+        """
+        @param robot: robot interface
+        @param timeout:
+        """
+        smach.State.__init__(self,
+                             outcomes=['done', 'abort', 'preempted'])
+        self.robot = robot
+        self.timeout = timeout
+        self.reset_head = reset_head
+
+    def execute(self, userdata=None):
+        # keep track of the duration
+        tstart = rospy.Time.now()
+
+        # Cancel head goal, we need it for navigation :)
+        if self.reset_head:
+            self.robot.head.close()
+
+        for part in self.robot.parts.itervalues():
+            # ensure all batteries are unplugged
+            if isinstance(part, Battery):
+                while part.is_charging:
+                    if tstart - rospy.Time.now() < rospy.Duration.from_sec(self.timeout):
```

Comment:
I am not sure this needs to be and `rospy.duration`. Maybe a float/int would also work

Reply:
First part is going to b e negative.... I don't think that should be the case. Also I prefer now > start+timeout